File: class/JavaScript/SyntaxBuilder.js
class ArgumentSyntaxBuilder {
    #build = "";
    #append(input) {
        if (!this.#build) {
            this.#build = input;
        } else {
            this.#build = `${this.#build} ${input}`;
        }
    }

    makeRegular(input, options) {
        if (typeof input !== "string") {
            throw new Error("SyntaxBuilder::makeRegular: Not a string");
        }

        const title = options?.title;
        if (title && typeof title !== "string") {
          throw new Error("SyntaxBuilder::makeRegular: Title not a string");
        }

        const inf = options?.inf;

        if (!options?.optional) {
            this.#append(`<${title ? `${title}: ` : ""}${input}${inf ? "..." : ""}>`);
        } else {
            this.#append(`<?${title ? `${title}: ` : ""}${input}${inf ? "..." : ""}>`);
        }

        return this;
    }

    makeChoice(inputs, options) {
        if (!Array.isArray(inputs)) {
            throw new Error("SyntaxBuilder::makeChoice: Not an array");
        }

        const def = options?.default;
        if (def && inputs.indexOf(def) === -1) {
            throw new Error("SyntaxBuilder::makeChoice: Default input not found");
        }

        if (options?.exactify) {
            inputs = inputs.map((el) => `"${el}"`);
        }

        const title = options?.title;
        if (title && typeof title !== "string") {
          throw new Error("SyntaxBuilder::makeRegular: Title not a string");
        }

        const inf = options?.inf;
        const splitString = inputs.join(" | ");

        if (!options?.optional) {
            this.#append(`<${title ? `${title}: ` : ""}${splitString}${def ? ` def = "${def}"` : ""}${inf ? "..." : ""}>`);
        } else {
            this.#append(`<?${title ? `${title}: ` : ""}${splitString}${def ? ` def = "${def}"` : ""}${inf ? "..." : ""}>`);
        }

        return this;
    }

    endBuild() {
        if (!this.#build) {
            throw new Error("SyntaxBuilder::endBuild: No build started");
        }

        const oldBuild = this.#build;
        this.#build = "";
        return oldBuild;
    }
}

const SyntaxBuilder = new ArgumentSyntaxBuilder();
const Command = ":eval";
const Syntax = SyntaxBuilder.makeRegular("JavaScript").endBuild();
console.log(Command + ` ${Syntax}`);

File: src/SyntaxBuilder.cs
using System.Linq;
using System.Collections.Generic;

namespace CS.ArgumentSyntaxBuilder
{
    public class ArgumentSyntaxBuilder
    {
        private string _Build = "";
        private void _Append(string Input)
        {
            if (String.IsNullOrEmpty(this._Build))
            {
                this._Build = Input;
            }
            else
            {
                this._Build = $"{this._Build} {Input}";
            }
        }

        public ArgumentSyntaxBuilder MakeRegular(string Input, ArgumentRegularOptions Options)
        {
            string Title = Options.Title;
            bool TitleExists = !String.IsNullOrEmpty(Title);

            string[]? Appensions = Options.Appensions;
            string JoinedAppensions = "";

            if (Appensions.Length != 0 && Array.Exists(Appensions, (el) => el.Length > 2))
            {
                throw new ArgumentException("Appensions cannot be more than 2 characters");
            }
            else if (Appensions.Length != 0)
            {
                JoinedAppensions = String.Join(",", (Appensions.Select((el) => el.ToLower())));
            }

            bool Inf = Options.Infinite;
            bool Opt = Options.Optional;
            bool AppensionsExist = !String.IsNullOrEmpty(JoinedAppensions);

            if (!Opt)
            {
                this._Append($"<{(TitleExists ? $"{Title}: " : "")}{Input}{(AppensionsExist ? $"[{JoinedAppensions}]" : "")}{(Inf ? "..." : "")}>");
            }
            else
            {
                this._Append($"<?{(TitleExists ? $"{Title}: " : "")}{Input}{(AppensionsExist ? $"[{JoinedAppensions}]" : "")}{(Inf ? "..." : "")}>");
            }

            return this;
        }

        public ArgumentSyntaxBuilder MakeChoice(string[] Inputs, ArgumentChoiceOptions Options)
        {
            string Title = Options.Title;
            bool TitleExists = !String.IsNullOrEmpty(Title);

            string Def = Options.Default;
            bool DefExists = !String.IsNullOrEmpty(Def);

            bool Inf = Options.Infinite;
            bool Opt = Options.Optional;
            bool Exactify = Options.Exactify;

            if (DefExists && Array.IndexOf(Inputs, Def) == -1)
            {
                throw new ArgumentException("Default Input not found in array");
            }

            if (Exactify)
            {
                Inputs = (Inputs.Select((el) => $"\"{el}\"")).ToArray();
            }

            string Joined = String.Join(" | ", Inputs);

            if (!Opt)
            {
                this._Append($"<{(TitleExists ? $"{Title}: " : "")}{Joined}{(DefExists ? $" def = \"{Def}\"" : "")}{(Inf ? "..." : "")}>");
            }
            else
            {
                this._Append($"<?{(TitleExists ? $"{Title}: " : "")}{Joined}{(DefExists ? $" def = \"{Def}\"" : "")}{(Inf ? "..." : "")}>");
            }

            return this;
        }

        public string EndBuild()
        {
            if (String.IsNullOrEmpty(this._Build))
            {
                throw new InvalidOperationException("No build started");
            }

            string OldBuild = this._Build;
            this._Build = "";
            return OldBuild;
        }
    }
}